flake.nix

```diff
@@ -0,0 +1,183 @@
+          sha256 = e2e-kind-node-sha256;
+          finalImageTag = "v${e2e-kind-node-version}";
+          finalImageName = "kindest/node";
+        };
+
+        cert-manager-controller-image = pkgs.dockerTools.pullImage{
+          imageName = "quay.io/jetstack/cert-manager-controller";
+          imageDigest = e2e-cert-manager-controller-digest;
+          sha256 = e2e-cert-manager-controller-sha256;
+          finalImageTag = e2e-cert-manager-version;
+          finalImageName = "quay.io/jetstack/cert-manager-controller";
+        };
+
+        cert-manager-webhook-image = pkgs.dockerTools.pullImage{
+          imageName = "quay.io/jetstack/cert-manager-webhook";
+          imageDigest = e2e-cert-manager-webhook-digest;
+          sha256 = e2e-cert-manager-webhook-sha256;
+          finalImageTag = e2e-cert-manager-version;
+          finalImageName = "quay.io/jetstack/cert-manager-webhook";
+        };
+
+        cert-manager-cainjector-image = pkgs.dockerTools.pullImage{
+          imageName = "quay.io/jetstack/cert-manager-cainjector";
+          imageDigest = e2e-cert-manager-cainjector-digest;
+          sha256 = e2e-cert-manager-cainjector-sha256;
+          finalImageTag = e2e-cert-manager-version;
+          finalImageName = "quay.io/jetstack/cert-manager-cainjector";
+        };
+
+        cert-manager-ctl-image = pkgs.dockerTools.pullImage{
+          imageName = "quay.io/jetstack/cert-manager-ctl";
+          imageDigest = e2e-cert-manager-ctl-digest;
+          sha256 = e2e-cert-manager-ctl-sha256;
+          finalImageTag = e2e-cert-manager-version;
+          finalImageName = "quay.io/jetstack/cert-manager-ctl";
+        };
+
+        csi-node-driver-registrar-image = pkgs.dockerTools.pullImage{
+          imageName = "quay.io/k8scsi/csi-node-driver-registrar";
+          imageDigest = csi-node-driver-registrar-digest;
+          sha256 = csi-node-driver-registrar-sha256;
+          finalImageTag = "v${csi-node-driver-registrar-version}";
+          finalImageName = "quay.io/k8scsi/csi-node-driver-registrar";
+        };
+
+        busybox-image = pkgs.dockerTools.pullImage{
+          imageName = "busybox";
+          imageDigest = e2e-busybox-digest;
+          sha256 = e2e-busybox-sha256;
+          finalImageTag = "latest";
+          finalImageName = "busybox";
+        };
+
+      in {
+        packages = {
+          default = containerImage;
+          container = containerImage;
+          inherit cert-manager-helm-chart;
+          inherit kind-node-image;
+          inherit cert-manager-controller-image;
+          inherit cert-manager-webhook-image;
+          inherit cert-manager-cainjector-image;
+          inherit cert-manager-ctl-image;
+          inherit cert-manager-csi;
+          inherit busybox-image;
+          inherit csi-node-driver-registrar-image;
+          inherit csi-lib-e2e;
+        };
+        devShells.default = pkgs.mkShell {
+          buildInputs = [
+            pkgs.kubectl
```

Comment:
How are these versioned/how do we select which version to use? It's currently pulling in a v1.23 binary but I don't see us specifying that anywhere